File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> umap;
        for (const int& num: nums) {
            umap[num]++;
        }

        vector<pair<int, int>> vec;
        for (const auto& keyVal : umap) {
            vec.push_back({keyVal.second, keyVal.first});
        }

        sort(vec.rbegin(), vec.rend());

        vector<int> ans;
        for (int i = 0; i < k; i++) {
            ans.push_back(vec[i].second);
        }
        return ans;
    }
};

File: Data Structures & Algorithms/buy-two-chocolates/submission-1.py
class Solution:
    def buyChoco(self, prices: List[int], money: int) -> int:
        # Using sort
        # prices.sort()
        # first, second = prices[0], prices[1]
        # ans = money - (first + second) 
        # return ans if ans >= 0 else money

        # Using greedy
        # Maintain min1 and min2

        min1 = min2 = money

        for p in prices:
            if p < min1:
                min2 = min1
                min1 = p
            elif p < min2:
                min2 = p
        ans = money - (min1 + min2) 
        return ans if ans >= 0 else money
        



        